QuizWebsite/src/problem/Abstract/AbstractAnswer.java

```diff
@@ -0,0 +1,67 @@
+package problem.Abstract;
+
+public abstract class AbstractAnswer {
+	private int grade;
+	private boolean graded = false;
+	private int testId;
+	private Object userAnswer;
+
+
+	/**
+	 *  Constructor stores test's id number and user's answer object.
+	 */
+	public AbstractAnswer(int newTestId, Object newUserAnswer) 
+	{
+		testId = newTestId;
+		userAnswer = newUserAnswer;
+	}
+
+	/**
+	 * Returns an object with all information about Answer.
+	 * @return answer.
+	 */
+	public Object getAnswer()
+	{	
+		return userAnswer;
+	}
+
+	/**
+	 * Sets the grade user in object.
+	 */
+	public void setGrade(int newGrade)
+	{
+		grade = newGrade;
+		graded = true;
+	}
+
+	/**
+	 * Returns id of the test.
+	 * @return id of test.
+	 */
+	public int getTestId()
+	{
+		return testId;
+	}
+
+	/**
+	 * Returns grade of user or null if answer was not graded yet.
+	 * @return grade or null.
+	 */
+	public Integer getGrade()

```

Comment:
Answer should not be responsible for its grade. It would be better if an answer asked quiz to evaluate it and store its grades.